src/renderers/dom/client/wrappers/ReactDOMInput.js

```diff
@@ -230,8 +230,24 @@ var ReactDOMInput = {
     // are not resetable nodes so this operation doesn't matter and actually
     // removes browser-default values (eg "Submit Query") when no value is
     // provided.
-    if (props.type !== 'submit' && props.type !== 'reset') {
-      node.value = node.value;
+
+    switch (props.type) {
+      case 'submit':
+      case 'reset':
+        break;
+      case 'date':
+      case 'datetime':
+      case 'datetime-local':
+      case 'month':
+      case 'time':
+      case 'week':
+        // this fixes the no-show issue on iOS Safari: https://github.com/facebook/react/issues/7233
+        node.value = '';
```

Comment:
Yes, though you've just now answered it. I am sorry to have forced a quick lecture on the DOM :).

I wanted to figure out if the delinking here eliminated the need for [the fix I added](https://github.com/facebook/react/pull/7359/files#diff-1b0e3b280991cd0b7cde194b30353b3eR211) to #7359 where `defaultValue` is only reassigned if it is different. 

[From a quick test outside of React](http://codepen.io/nhunzaker/pen/jAvxvQ), I thought this PR would make it unnecessary. However it isn't, and I became confused. It looks like, [no matter what, `value` and `checked` get assigned in `ReactDOMInput.getHostProps`](https://github.com/facebook/react/blob/master/src/renderers/dom/client/wrappers/ReactDOMInput.js#L73-L74), and subsequently get sent out as DOM mutations. Even for uncontrolled inputs.

So that's what I'm working through now. No need to continue talking about it here, but I'd like to avoid being able to do the comparison check  and wanted to better understand the solution in this PR.